### PR TITLE
Support deletetalk parameter in WikiPage::delete() and WikiFile::delete()

### PR DESCRIPTION
**Pull request by Xymph**
This adds support for deleting corresponding talk pages when deleting pages and files (which was introduced in MediaWiki v1.38), using the version-dependent mechanism added in #151 .

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and [Keep a Changelog](http://keepachangelog.com/).
 ### Added
 
 - Added internal mechanism to handle version-dependent parameters of API modules ([#151])
+- Support version-dependent `deletetalk` parameter in `WikiPage::delete()` and `WikiFile::delete()` ([#152])
```
Comment (waldyrious) [inline CHANGELOG.md:13]: Shouldn't there also be a similar changelog entry for the logout method? :thinking:

File: Wikimate.php [hunk 2]
```diff
@@ -1419,10 +1420,11 @@ public function newSection($name, $text)
...
      * Deletes the page.
      *
-     * @param   string   $reason  Reason for the deletion
-     * @return  boolean           True if page was deleted successfully
+     * @param   string   $reason   Reason for the deletion
+     * @param   boolean  $talktoo  True to delete talk page too (MW 1.38+)
+     * @return  boolean            True if page was deleted successfully
      */
-    public function delete($reason = null)
+    public function delete($reason = null, $talktoo = false)
```
Comment (waldyrious) [inline Wikimate.php:1427]: I find it odd that [the default in MediaWiki is false](https://github.com/wikimedia/mediawiki/blob/1.38.0/includes/api/ApiDelete.php#L175). Would you be open to deviate from that behavior and default to deleting the talk page too? I can't think of many cases where one would want to delete the main page but keep its talk page around, but maybe I just lack imagination! :sweat_smile:

File: Wikimate.php [hunk 3]
```diff
@@ -1432,6 +1434,10 @@ public function delete($reason = null)
         if (!is_null($reason)) {
             $data['reason'] = $reason;
         }
+        // Check if deletetalk parameter is supported (it is since MediaWiki v1.38)
```
Comment (waldyrious) [inline Wikimate.php:1437]:
```suggestion
        // Check if the `deletetalk` parameter is supported (it is since MediaWiki v1.38)
```

(Does markdown code formatting work in the rendered docs?)

File: Wikimate.php [hunk 4]
```diff
@@ -2334,6 +2341,10 @@ public function delete($reason = null, $archivename = null)
         if (!is_null($archivename)) {
             $data['oldimage'] = $archivename;
         }
+        // Check if deletetalk parameter is supported (it is since MediaWiki v1.38)
```
Comment (waldyrious) [inline Wikimate.php:2344]:
```suggestion
        // Check if the `deletetalk` parameter is supported (it is since MediaWiki v1.38)
```

File: Wikimate.php (hunk 3 above)
Comment (Xymph) [reply Wikimate.php:1437]: Uhm, inline comments are not included in the rendered docs, so the backticks are not useful. And to keep the line length close to 80 characters (for undeletetalk in the next PR too), I think it's fine to leave out "the" here. It's just a comment after all. :)

File: CHANGELOG.md (hunk 1 above)
Comment (Xymph) [reply CHANGELOG.md:13]: Oh, right. Added.

File: Wikimate.php (hunk 2 above)
Comment (Xymph) [reply Wikimate.php:1427]: Historically talk pages were not deleted while deleting a main/file space page, this required a separate talk delete, so the default makes sense. And a common situation (at least on the Doom Wiki) is that the talk page has a discussion/vote about the article deletion, which is then preserved to document the voting result. So I definitely don't think defaulting to `true` is a good angle for Wikimate.

File: Wikimate.php (hunk 3 above)
Comment (waldyrious) [reply Wikimate.php:1437]: Uh, silly me, I clearly wasn't paying attention. Ok, fair enough, let's disregard this suggestion.

File: Wikimate.php (hunk 2 above)
Comment (waldyrious) [reply Wikimate.php:1427]: Ah, that does make sense, yeah. Ok then, let's follow suit then.

File: CHANGELOG.md [hunk 5]
```diff
@@ -23,6 +24,8 @@ and [Keep a Changelog](http://keepachangelog.com/).
 ### Fixed
 
 - Fixed example not using correct array key for invalid login errors ([#148])
+- Prevented warning in `Wikimate::logout()` about `token` parameter that was unsupported
+  before MediaWiki v1.34 ([#151])
```
Comment (waldyrious) [inline CHANGELOG.md:28]: I'd have preferred this to be in a separate commit (ideally it would have been part of the previous PR, but that ship has sailed). However, it's such a minor thing that I don't think it's worth the hassle to address now.

**Review by waldyrious — APPROVED** [review]
LGTM!
